pkg/manifests/configaware.go

```diff
@@ -208,18 +210,14 @@ func (ca *configAware) getResourcesByID(ctx context.Context) (map[string]resourc
 		if err != nil {
 			return nil, err
 		}
-		relConfigFilePath, err := cf.RelativeConfigPath()
-		if err != nil {
-			return nil, err
-		}
-		resources, err := ca.manifests.ParseManifest(resourceManifests, relConfigFilePath)
+		resources, err := ca.manifests.ParseManifest(resourceManifests, cf.Source())
```

Comment:
Passing the config file's source context instead of simply the `.flux.yaml` filepath will break [this check](https://github.com/fluxcd/flux/blob/859d7112c619a3f4c96fbc8d3e3bab29366ad687/pkg/daemon/sync.go#L224)

Reply:
I guess we can make the `Source()` method of resources a bit more sophisticated, separating context and filepath.

Or, we could add an extra `IsGenerated()` method for resources.

And/Or, we should probably reconsider always adding the generated resources to the commit log (which is the only place where I believe `Source()` is expected to follow certain structure)

Reply:
So it does. I have touched that code but I'd forgotten that it relied on `Source` for something other than presentation. (Possibly that is the case elsewhere too)

That is a shame; I either have to figure out how to reform that bit of code (in the smaller sense of not relying on the Source there, or in the larger sense of fixing how changed files are calculated); or, hack another thing into resources so that error messages have access to the extra bit of info.

Reply:
Of the alternatives (thanks for thinking those through), I prefer keeping Source for presentation only; which means either omitting generated resources from the event log, or calculating it some other way.

Reply:
Perhaps for the purpose of making error messages better I could find Another Way that is neutral with respect to `.Source`.

Reply:
Sounds good!

Reply:
Opened https://github.com/fluxcd/flux/issues/2601 for the bigger problem with that sync.go code.

Reply:
I've gone back to using the config file path as the source for generated manifests, so it won't affect the bit of code you spotted @2opremio .